components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/PushedAuthDataStore.java

```diff
@@ -0,0 +1,208 @@
+    private static final String SQL_DELETE_EXPIRED_DATA_TASK_DB2SQL =
+            "DELETE FROM IDN_OAUTH_PAR WHERE REQ_URI_REF IN " +
+                    "(SELECT REQ_URI_REF FROM IDN_OAUTH_PAR WHERE SCHEDULED_EXPIRY < ? FETCH FIRST %d ROWS ONLY)";
+    private static final String MYSQL_DATABASE = "MySQL";
+    private static final String MARIA_DATABASE = "MariaDB";
+    private static final String H2_DATABASE = "H2";
+    private static final String DB2_DATABASE = "DB2";
+    private static final String MS_SQL_DATABASE = "MS SQL";
+    private static final String MICROSOFT_DATABASE = "Microsoft";
+    private static final String POSTGRESQL_DATABASE = "PostgreSQL";
+    private static final String INFORMIX_DATABASE = "Informix";
+    private static final int DEFAULT_DELETE_LIMIT = 50000;
+    private static final int EXPIRATION_GRACE_PERIOD_IN_MINUTES = 10;
+    private static final int DELETE_CHUNK_SIZE = DEFAULT_DELETE_LIMIT;
+    private static final String CLEAN_UP_PERIOD_DEFAULT_IN_MINUTES = "1440";
+    private String sqlDeleteExpiredDataTask;
+    private boolean requestCleanupEnabled = true;
+
+    /**
+     * Gets the instance of PushedAuthDataStore.
+     *
+     * @return Instance of PushedAuthDataStore.
+     */
+    public static PushedAuthDataStore getInstance() {
+        if (instance == null) {
+            synchronized (PushedAuthDataStore.class) {
+                if (instance == null) {
+                    instance = new PushedAuthDataStore();
+                }
+            }
+        }
+        return instance;
+    }
+
+    private PushedAuthDataStore() {
+
+        String isCleanUpEnabledVal
+                = IdentityUtil.getProperty("JDBCPersistenceManager.PushedAuthReqCleanUp.Enable");
+        if (StringUtils.isNotBlank(isCleanUpEnabledVal)) {
+            requestCleanupEnabled = Boolean.parseBoolean(isCleanUpEnabledVal);
+        }
+        if (requestCleanupEnabled) {
+            long requestCleanupPeriod = getCleanUpPeriod();
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("PAR clean up task enabled to run in %d minutes intervals",
+                        requestCleanupPeriod));
+            }
+            PushedAuthReqCleanupService pushedAuthReqCleanupService =
+                    new PushedAuthReqCleanupService(requestCleanupPeriod / 4, requestCleanupPeriod);
+            pushedAuthReqCleanupService.activateCleanUp();
+        }
+    }
+
+    /**
+     * Gets the DB specific query for the request removal task.
+     *
+     * @return DB specific query for the request removal task.
+     * @throws IdentityApplicationManagementException Error when retrieving DB connection meta-data.
+     */
+    private String getDBSpecificRequestRemovalQuery() throws IdentityApplicationManagementException {
+
+        Connection connection = null;
+        try {
+            connection = IdentityDatabaseUtil.getDBConnection(true);
```

Comment:
Can we use try-with-resources

Reply:
If we use that we cannot use the connection inside the catch clause eg: `IdentityDatabaseUtil.rollbackTransaction(connection);`  since it's only available within the try block